### ISSUE TITLE
Running with HCL JSON fails

### ISSUE THREAD

**Issue by alikhajeh1** [infracost/infracost#801]
Repro steps:
1. use https://www.hcl2json.com/ to convert `examples/terraform/main.tf` to a json file
2. create a new directory called `jsontest` and inside it create a new `main.tf.json` file with the above content.
3. run `infracost breakdown --path jsontest` and `infracost breakdown --path jsontest/main.tf.json` both fail with:
```
Error: Could not detect path type for jsontest

Use the --path flag to specify the path to one of the following:
 - Terraform plan JSON file
 - Terraform directory
 - Terraform plan file
 - Terraform state JSON file
```

Running `terraform init` inside the `jsontest` dir, then running `infracost breakdown --path jsontest` works as expected.

Expected behavior: infracost could running `terraform init` like it does for `Terraform directory` and work without the user having to know the difference between the directory containing .tf or .tf.json files.

A user reported this issue, their dir structure is:
<img width="380" alt="test" src="https://user-images.githubusercontent.com/638577/121417441-ce97d600-c961-11eb-82f5-8047e75aeff1.png">

**Comment by aliscott** [comment]
I think adding `.tf.json` to https://github.com/infracost/infracost/blob/65f9c7d43e3e13611272ccfe8a9d6935d36532ac/internal/providers/terraform/dir_provider.go#L347 should fix this.

**Comment by alikhajeh1** [comment]
Released in v0.9.1
